lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -144,24 +144,28 @@ public byte[] Call(byte[] memento)
 
         /// <summary>
         /// Task close handler.
-        /// If the closed event is sent from driver, set _shouldCloseTask to 1 so that to inform the Call() to stop at the end of the current iteration.
+        /// Set _shouldCloseTask to 1 so that to inform the Call() to stop at the end of the current iteration.
         /// Then waiting for the signal from Call method. Either it is signaled or after _enforceCloseTimeoutMilliseconds,
-        /// checks if the task has been stopped. If not, throw IMRUTaskSystemException to enforce the task to stop.
+        /// If the closed event is sent from driver, checks if the task has been stopped. If not, throw IMRUTaskSystemException 
+        /// to enforce the task to stop.
         /// </summary>
         /// <param name="closeEvent"></param>
         public void OnNext(ICloseEvent closeEvent)
         {
-            var msg = Encoding.UTF8.GetString(closeEvent.Value.Value);
-            if (closeEvent.Value.IsPresent() && msg.Equals(TaskManager.CloseTaskByDriver))
-            {
-                Logger.Log(Level.Info, "The task received close event with message: {0}.", msg);
-                Interlocked.Exchange(ref _shouldCloseTask, 1);
-
-                _waitToCloseEvent.Wait(TimeSpan.FromMilliseconds(_enforceCloseTimeoutMilliseconds));
+            Interlocked.Exchange(ref _shouldCloseTask, 1);
+            _waitToCloseEvent.Wait(TimeSpan.FromMilliseconds(_enforceCloseTimeoutMilliseconds));
 
-                if (Interlocked.Read(ref _isTaskStopped) == 0)
+            if (closeEvent.Value.IsPresent())
+            {
+                var msg = Encoding.UTF8.GetString(closeEvent.Value.Value);
+                if (msg.Equals(TaskManager.CloseTaskByDriver))
                 {
-                    throw new IMRUTaskSystemException(TaskManager.TaskKilledByDriver);
+                    Logger.Log(Level.Info, "The task received close event with message: {0}.", msg);
+
+                    if (Interlocked.Read(ref _isTaskStopped) == 0)
```

Comment:
No, but you set it to 1 in line 155. The code will never reach here without passing line 155.